--- /opt/phpenv/versions/7.0/bin/phpize7.0	2017-04-19 20:28:38.939093808 +0300
+++ /opt/phpenv/versions/7.0/bin/phpize7.0	2017-04-19 20:29:05.146788581 +0300
@@ -9,7 +9,9 @@ includedir="`eval echo ${prefix}/include
 builddir="`pwd`"
 SED="/bin/sed"
 
-FILES_BUILD="mkdep.awk scan_makefile_in.awk shtool libtool.m4 ax_check_compile_flag.m4"
+aclocaldir="/usr/share/aclocal"
+FILES_BUILD="mkdep.awk scan_makefile_in.awk shtool libtool.m4 ax_check_compile_flag.m4"
+LIBTOOL_FILES="libtool.m4 ltoptions.m4 ltsugar.m4 ltversion.m4 lt~obsolete.m4"
 FILES="acinclude.m4 Makefile.global config.sub config.guess ltmain.sh run-tests*.php"
 CLEAN_FILES="$FILES *.o *.lo *.la .deps .libs/ build/ modules/ install-sh \
 	mkinstalldirs missing config.nice config.sub config.guess configure configure.in \
