--- configure	2017-12-21 20:33:18.972354408 +0300
+++ configure	2017-12-21 20:34:35.676495586 +0300
@@ -1967,7 +1967,7 @@ Some influential environment variables:
               you have headers in a nonstandard directory <include dir>
   CPP         C preprocessor
   YACC        The `Yet Another Compiler Compiler' implementation to use.
-              Defaults to the first program found out of: `bison -y', `byacc',
+              Defaults to the first program found out of: `/opt/bison/bin/bison -y', `byacc',
               `yacc'.
   YFLAGS      The list of arguments that will be passed by default to $YACC.
               This script will default YFLAGS to the empty string to avoid a
@@ -5152,7 +5152,7 @@ $as_echo "no" >&6; }
 
 
 
-  for ac_prog in 'bison -y' byacc
+  for ac_prog in '/opt/bison/bin/bison -y' byacc
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -5210,7 +5210,7 @@ if ${php_cv_bison_version+:} false; then
   $as_echo_n "(cached) " >&6
 else
 
-      bison_version_vars=`bison --version 2> /dev/null | grep 'GNU Bison' | cut -d ' ' -f 4 | $SED -e 's/\./ /' | tr -d a-z`
+      bison_version_vars=`/opt/bison/bin/bison --version 2> /dev/null | grep 'GNU Bison' | cut -d ' ' -f 4 | $SED -e 's/\./ /' | $SED -e 's/\./ /' | tr -d a-z`
       php_cv_bison_version=invalid
       if test -n "$bison_version_vars"; then
         set $bison_version_vars
@@ -104915,7 +104915,7 @@ if ${php_cv_bison_version+:} false; then
   $as_echo_n "(cached) " >&6
 else
 
-      bison_version_vars=`bison --version 2> /dev/null | grep 'GNU Bison' | cut -d ' ' -f 4 | $SED -e 's/\./ /' | tr -d a-z`
+      bison_version_vars=`/opt/bison/bin/bison --version 2> /dev/null | grep 'GNU Bison' | cut -d ' ' -f 4 | $SED -e 's/\./ /' | $SED -e 's/\./ /' | tr -d a-z`
       php_cv_bison_version=invalid
       if test -n "$bison_version_vars"; then
         set $bison_version_vars
