Index: php53-5.3.29/src/configure
===================================================================
--- php53-5.3.29.orig/src/configure	2016-10-06 17:21:32.038609490 +0300
+++ php53-5.3.29/src/configure	2016-10-06 17:21:32.035609582 +0300
@@ -88985,13 +88985,13 @@
 
 echo $ac_n "checking for SHA512 crypt""... $ac_c" 1>&6
 echo "configure:88988: checking for SHA512 crypt" >&5
-if eval "test \"`echo '$''{'ac_cv_crypt_SHA512'+set}'`\" = set"; then
+if eval "test \"`echo '$''{'ac_cv_crypt_sha512'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   
 if test "$cross_compiling" = yes; then
   
-  ac_cv_crypt_SHA512=no
+  ac_cv_crypt_sha512=no
 
 else
   cat > conftest.$ac_ext <<EOF
@@ -89022,14 +89022,14 @@
 if { (eval echo configure:89023: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
 then
   
-  ac_cv_crypt_SHA512=yes
+  ac_cv_crypt_sha512=yes
 
 else
   echo "configure: failed program was:" >&5
   cat conftest.$ac_ext >&5
   rm -fr conftest*
   
-  ac_cv_crypt_SHA512=no
+  ac_cv_crypt_sha512=no
 
 fi
 rm -fr conftest*
@@ -89037,17 +89037,17 @@
 
 fi
 
-echo "$ac_t""$ac_cv_crypt_SHA512" 1>&6
+echo "$ac_t""$ac_cv_crypt_sha512" 1>&6
 
 echo $ac_n "checking for SHA256 crypt""... $ac_c" 1>&6
 echo "configure:89044: checking for SHA256 crypt" >&5
-if eval "test \"`echo '$''{'ac_cv_crypt_SHA256'+set}'`\" = set"; then
+if eval "test \"`echo '$''{'ac_cv_crypt_sha256'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   
 if test "$cross_compiling" = yes; then
   
-  ac_cv_crypt_SHA256=no
+  ac_cv_crypt_sha256=no
 
 else
   cat > conftest.$ac_ext <<EOF
@@ -89078,14 +89078,14 @@
 if { (eval echo configure:89079: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
 then
   
-  ac_cv_crypt_SHA256=yes
+  ac_cv_crypt_sha256=yes
 
 else
   echo "configure: failed program was:" >&5
   cat conftest.$ac_ext >&5
   rm -fr conftest*
   
-  ac_cv_crypt_SHA256=no
+  ac_cv_crypt_sha256=no
 
 fi
 rm -fr conftest*
@@ -89093,7 +89093,7 @@
 
 fi
 
-echo "$ac_t""$ac_cv_crypt_SHA256" 1>&6
+echo "$ac_t""$ac_cv_crypt_sha256" 1>&6
 
 
 if test "$ac_cv_crypt_blowfish" = "no" || test "$ac_cv_crypt_des" = "no" || test "$ac_cv_crypt_ext_des" = "no" || test "x$php_crypt_r" = "x0"; then
Index: php53-5.3.29/src/ext/standard/config.m4
===================================================================
--- php53-5.3.29.orig/src/ext/standard/config.m4	2016-10-06 17:21:32.038609490 +0300
+++ php53-5.3.29/src/ext/standard/config.m4	2016-10-06 17:21:32.036609551 +0300
@@ -172,7 +172,7 @@
   ac_cv_crypt_blowfish=no
 ])])
 
-AC_CACHE_CHECK(for SHA512 crypt, ac_cv_crypt_SHA512,[
+AC_CACHE_CHECK(for SHA512 crypt, ac_cv_crypt_sha512,[
 AC_TRY_RUN([
 #if HAVE_UNISTD_H
 #include <unistd.h>
@@ -194,14 +194,14 @@
 	exit(0);
 #endif
 }],[
-  ac_cv_crypt_SHA512=yes
+  ac_cv_crypt_sha512=yes
 ],[
-  ac_cv_crypt_SHA512=no
+  ac_cv_crypt_sha512=no
 ],[
-  ac_cv_crypt_SHA512=no
+  ac_cv_crypt_sha512=no
 ])])
 
-AC_CACHE_CHECK(for SHA256 crypt, ac_cv_crypt_SHA256,[
+AC_CACHE_CHECK(for SHA256 crypt, ac_cv_crypt_sha256,[
 AC_TRY_RUN([
 #if HAVE_UNISTD_H
 #include <unistd.h>
@@ -224,11 +224,11 @@
 	exit(0);
 #endif
 }],[
-  ac_cv_crypt_SHA256=yes
+  ac_cv_crypt_sha256=yes
 ],[
-  ac_cv_crypt_SHA256=no
+  ac_cv_crypt_sha256=no
 ],[
-  ac_cv_crypt_SHA256=no
+  ac_cv_crypt_sha256=no
 ])])
 
 
