--- a/src/sapi/apache2handler/config.m4
+++ b/src/sapi/apache2handler/config.m4
@@ -38,7 +38,7 @@
   APXS_BINDIR=`$APXS -q BINDIR`
   APXS_HTTPD=`$APXS -q SBINDIR`/`$APXS -q TARGET`
   APXS_CFLAGS=`$APXS -q CFLAGS`
-  APXS_MPM=`$APXS -q MPM_NAME`
+  APXS_MPM=`$APXS -q MPM_NAME >/dev/null`
   APU_BINDIR=`$APXS -q APU_BINDIR`
   APR_BINDIR=`$APXS -q APR_BINDIR`
 
@@ -60,12 +60,16 @@
   APACHE_CFLAGS="$APACHE_CPPFLAGS -I$APXS_INCLUDEDIR $APR_CFLAGS $APU_CFLAGS"
 
 dnl  # Test that we're trying to configure with apache 2.x
-dnl  PHP_AP_EXTRACT_VERSION($APXS_HTTPD)
+  PHP_AP_EXTRACT_VERSION($APXS_HTTPD)
 dnl  if test "$APACHE_VERSION" -le 2000000; then
 dnl    AC_MSG_ERROR([You have enabled Apache 2 support while your server is Apache 1.3.  Please use the appropiate switch --with-apxs (without the 2)])
 dnl  elif test "$APACHE_VERSION" -lt 2000044; then
 dnl    AC_MSG_ERROR([Please note that Apache version >= 2.0.44 is required])
 dnl  fi
+  if test $APACHE_VERSION -ge 2004000; then
+    AC_MSG_RESULT([Apache version >= 2.4.0 - force prefork build. Sorry.])
+    APXS_MPM=prefork
+  fi
 
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
   if test -z `$APXS -q SYSCONFDIR`; then
--- a/src/configure
+++ b/src/configure
@@ -5618,7 +5618,7 @@
   APXS_BINDIR=`$APXS -q BINDIR`
   APXS_HTTPD=`$APXS -q SBINDIR`/`$APXS -q TARGET`
   APXS_CFLAGS=`$APXS -q CFLAGS`
-  APXS_MPM=`$APXS -q MPM_NAME`
+  APXS_MPM=`$APXS -q MPM_NAME 2>/dev/null`
   APU_BINDIR=`$APXS -q APU_BINDIR`
   APR_BINDIR=`$APXS -q APR_BINDIR`
 
@@ -5650,10 +5650,13 @@
 
   APACHE_VERSION=`expr $4 \* 1000000 + $5 \* 1000 + $6`
 
-  if test "$APACHE_VERSION" -le 2000000; then
+  if test $APACHE_VERSION -le 2000000; then
     { echo "configure: error: You have enabled Apache 2 support while your server is Apache 1.3.  Please use the appropiate switch --with-apxs (without the 2)" 1>&2; exit 1; }
-  elif test "$APACHE_VERSION" -lt 2000040; then
-    { echo "configure: error: Please note that Apache version >= 2.0.40 is required" 1>&2; exit 1; }
+  elif test $APACHE_VERSION -lt 2000044; then
+    { echo "configure: error: Please note that Apache version >= 2.0.44 is required" 1>&2; exit 1; }
+  fi
+  if test $APACHE_VERSION -ge 2004000; then
+    { echo "configure: Apache version >= 2.4.0 - force prefork build. Sorry." 1>&2; APXS_MPM=prefork; }
   fi
 
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
@@ -6466,7 +6469,7 @@
   APXS_BINDIR=`$APXS -q BINDIR`
   APXS_HTTPD=`$APXS -q SBINDIR`/`$APXS -q TARGET`
   APXS_CFLAGS=`$APXS -q CFLAGS`
-  APXS_MPM=`$APXS -q MPM_NAME`
+  APXS_MPM=`$APXS -q MPM_NAME 2>/dev/null`
   APU_BINDIR=`$APXS -q APU_BINDIR`
   APR_BINDIR=`$APXS -q APR_BINDIR`
 
@@ -6498,11 +6501,14 @@
 
   APACHE_VERSION=`expr $4 \* 1000000 + $5 \* 1000 + $6`
 
-  if test "$APACHE_VERSION" -le 2000000; then
+  if test $APACHE_VERSION -le 2000000; then
     { echo "configure: error: You have enabled Apache 2 support while your server is Apache 1.3.  Please use the appropiate switch --with-apxs (without the 2)" 1>&2; exit 1; }
-  elif test "$APACHE_VERSION" -lt 2000044; then
+  elif test $APACHE_VERSION -lt 2000044; then
     { echo "configure: error: Please note that Apache version >= 2.0.44 is required" 1>&2; exit 1; }
   fi
+  if test $APACHE_VERSION -ge 2004000; then
+    { echo "configure: Apache version >= 2.4.0 - force prefork build. Sorry." 1>&2; APXS_MPM=prefork; }
+  fi
 
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
   if test -z `$APXS -q SYSCONFDIR`; then
