diff -up php-5.2.17/main/php_variables.c.bug-61000 php-5.2.17/main/php_variables.c
--- php-5.2.17/src/main/php_variables.c.bug-61000	2012-05-03 22:41:42.000000000 -0400
+++ php-5.2.17/src/main/php_variables.c	2012-05-03 22:44:39.000000000 -0400
@@ -136,11 +136,12 @@ PHPAPI void php_register_variable_ex(cha
 
 				if (track_vars_array) {
 					ht = Z_ARRVAL_P(track_vars_array);
+					zend_symtable_del(ht, var, var_len + 1);
 				} else if (PG(register_globals)) {
 					ht = EG(active_symbol_table);
+					zend_symtable_del(ht, var, var_len + 1);
 				}
 
-				zend_hash_del(ht, var, var_len + 1);
 				zval_dtor(val);
 
 				/* do not output the error message to the screen,
