diff -up php-5.2.17/ext/curl/interface.c.bug-62499 php-5.2.17/ext/curl/interface.c
--- php-5.2.17/src/ext/curl/interface.c.bug-62499	2012-08-22 09:35:34.000000000 +0700
+++ php-5.2.17/src/ext/curl/interface.c	2012-08-22 09:36:04.000000000 +0700
@@ -1720,7 +1720,7 @@ static int _php_curl_setopt(php_curl *ch
 
 			convert_to_string_ex(zvalue);
 
-			if (!Z_STRLEN_PP(zvalue) || php_check_open_basedir(Z_STRVAL_PP(zvalue) TSRMLS_CC) || (PG(safe_mode) && !php_checkuid(Z_STRVAL_PP(zvalue), "rb+", CHECKUID_CHECK_MODE_PARAM))) {
+			if ((Z_STRLEN_PP(zvalue) && php_check_open_basedir(Z_STRVAL_PP(zvalue) TSRMLS_CC)) || (PG(safe_mode) && !php_checkuid(Z_STRVAL_PP(zvalue), "rb+", CHECKUID_CHECK_MODE_PARAM))) {
 				RETVAL_FALSE;
 				return 1;
 			}
