diff -up php-5.2.17/ext/standard/basic_functions.c.bug-60222 php-5.2.17/ext/standard/basic_functions.c
--- php-5.2.17/src/ext/standard/basic_functions.c.bug-60222	2012-05-03 23:29:11.000000000 -0400
+++ php-5.2.17/src/ext/standard/basic_functions.c	2012-05-03 23:30:20.000000000 -0400
@@ -4825,6 +4825,15 @@ PHP_FUNCTION(time_nanosleep)
 		return;
 	}
 
+	if (tv_sec < 0) {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "The seconds value must be greater than 0");
+		RETURN_FALSE;
+	}
+	if (tv_nsec < 0) {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "The nanoseconds value must be greater than 0");
+		RETURN_FALSE;
+	}
+
 	php_req.tv_sec = (time_t) tv_sec;
 	php_req.tv_nsec = tv_nsec;
 	if (!nanosleep(&php_req, &php_rem)) {
