diff -up php-5.2.17/ext/standard/array.c.bug-61730 php-5.2.17/ext/standard/array.c
--- php-5.2.17/src/ext/standard/array.c.bug-61730	2012-07-23 02:01:59.000000000 -0400
+++ php-5.2.17/src/ext/standard/array.c	2012-07-23 02:05:50.000000000 -0400
@@ -1157,7 +1157,6 @@ static int php_array_walk(HashTable *tar
 	char  *string_key;
 	uint   string_key_len;
 	ulong  num_key;
-	HashPosition pos;
 	zend_fcall_info_cache array_walk_fci_cache = empty_fcall_info_cache;
 
 	/* Set up known arguments */
@@ -1167,10 +1166,9 @@ static int php_array_walk(HashTable *tar
 		(*userdata)->refcount++;
 	}
 
-	zend_hash_internal_pointer_reset_ex(target_hash, &pos);
-
 	/* Iterate through hash */
-	while (!EG(exception) && zend_hash_get_current_data_ex(target_hash, (void **)&args[0], &pos) == SUCCESS) {
+	zend_hash_internal_pointer_reset(target_hash);
+	while (!EG(exception) && zend_hash_get_current_data(target_hash, (void **)&args[0]) == SUCCESS) {
 		if (recursive && Z_TYPE_PP(args[0]) == IS_ARRAY) {
 			HashTable *thash;
 			
@@ -1193,7 +1191,7 @@ static int php_array_walk(HashTable *tar
 			MAKE_STD_ZVAL(key);
 
 			/* Set up the key */
-			if (zend_hash_get_current_key_ex(target_hash, &string_key, &string_key_len, &num_key, 0, &pos) == HASH_KEY_IS_LONG) {
+			if (zend_hash_get_current_key_ex(target_hash, &string_key, &string_key_len, &num_key, 0, NULL) == HASH_KEY_IS_LONG) {
 				Z_TYPE_P(key) = IS_LONG;
 				Z_LVAL_P(key) = num_key;
 			} else {
@@ -1236,7 +1234,7 @@ static int php_array_walk(HashTable *tar
 			zval_ptr_dtor(&key);
 			key = NULL;
 		}
-		zend_hash_move_forward_ex(target_hash, &pos);
+		zend_hash_move_forward(target_hash);
 	}
 	
 	if (userdata) {
